src/app.ts

```diff
@@ -64,29 +69,64 @@ async function app() {
       'exit with error if this is not the latest version of SUSHI',
       false
     )
-    .option('-i, --init', 'initialize a SUSHI project')
-    .option('-u, --update-dependencies', 'update FHIR packages in project configuration')
-    .version(getVersion(), '-v, --version', 'print SUSHI version')
+    .action(async function (projectPath, options) {
+      await runBuild(projectPath, options);
+    })
     .on('--help', () => {
       console.log('');
       console.log('Additional information:');
       console.log('  [path-to-fsh-project]');
       console.log('    Default: "."');
       console.log('  -o, --out <out>');
       console.log('    Default: "fsh-generated"');
+    });
+
+  program
+    .command('updateDependencies [path-to-fsh-project]')
+    .description('update FHIR packages in project configuration')
+    .action(async function (projectPath) {
+      const input = ensureInputDir(projectPath);
+      const config: Configuration = readConfig(input);
+      await updateExternalDependencies(config);
+      process.exit(0);
     })
-    .arguments('[path-to-fsh-defs]')
-    .action(function (pathToFshDefs) {
-      input = pathToFshDefs;
-    })
-    .showHelpAfterError()
-    .parse(process.argv)
-    .opts();
+    .on('--help', () => {
+      console.log('');
+      console.log('Additional information:');
+      console.log('  [path-to-fsh-project]');
+      console.log('    Default: "."');
+    });
 
-  if (program.init) {
-    await init();
-    process.exit(0);
+  program
+    .command('init')
+    .description('initialize a SUSHI project')
+    .action(async function () {
+      await init();
+      process.exit(0);
+    });
+
+  // Maintain backwards compatibility with prior SUSHI command structure
+  if (
+    process.argv.length === 2 ||
+    (process.argv.length > 2 && !SUPPORTED_COMMANDS.includes(process.argv[2]))
+  ) {
+    if (process.argv.some(a => a === '-i' || a === '--init')) {
+      // init was moved to a separate command, so log a message to indicate how to use it
+      console.log(
+        'The --init option has been moved to a separate command. Instead, run the following command: sushi init'
+      );
+      process.exit(0);
+    } else if (process.argv.length === 2 || process.argv.slice(2).some(a => fs.existsSync(a))) {
```

Comment:
This, unfortunately, doesn't support running SUSHI w/ an argument but no path.  E.g., this no longer works:
```
sushi --debug
```

I think we can support this by adding another condition in the `else if` above, checking if all args after index 1 start with `-`.

Reply:
Oh yep I forgot about that case.

Reply:
So this is a fun one! I was not super loving how many conditional cases we were trying to handle, so I looked into it a little, talked to @mgramigna, and learned that there is an `isDefault` option on commands (see [Commands](https://github.com/tj/commander.js/#commands)). So I removed all the handling of processing the arguments and I removed the hardcoded list of options (yay!), and instead just used `{ isDefault: true }` on the `build` command. That should resolve the case you pointed out (e.g. `sushi --debug`) and it should more cleanly handle all the cases the if block was trying (and not really succeeding) to catch.

I also added in an option for `--init`, hid it from the `build` help text, and added in special case logic for logging the error message for that command. I think it works as we want it to, but I don't think this is a super necessary message because the IGP doesn't use that option, and this is a breaking change, so if it doesn't work or you don't like the extra bulk it adds, I'm happy to remove it.

Reply:
LGTM

Reply:
Ugh so I did realize that since this just automatically runs `build`, the check for if something is an actual directory is no longer included, so if you ran `sushi inti` (or some other typo), it will try to run `build` with a directory `inti`. I'm not sure if this is worth going back to the old approach with the multiple conditionals, but I feel like it's worth pointing out before decide if we definitely want to keep the `isDefault` approach.

Reply:
I don't know if this adds more trouble than its worth, but could you make it so that one of the first things `build` does is check `argv[1]`?  If it's `build` then the user invoked it directly, so let it do its thing.  If it is not `build` and it does not start with `-`, then check if it points to an existing file.  If it doesn't, print SUSHI usage instructions?

Reply:
I think that makes sense, and I've attempted it. I'm curious what you think about it. I had originally been excited about this, but I want to make sure my changes are actually solving more problems than the original approach.

The new commit with the handling in `build` is [45fd3ec](https://github.com/FHIR/sushi/pull/1163/commits/45fd3ec80f313b16b64900b8891681d529c48107). This commit plus [9e1ecc6](https://github.com/FHIR/sushi/pull/1163/commits/9e1ecc6d0068d3b6125f8829267c5997185eeed1) rework the `build` command to hopefully allow the old way of running SUSHI while still scoping it enough to provide useful information when people are trying to use the new commands but doing something wrong. Let me know what you think.